src/nerpa_pipeline/NRPSPredictor_utils/json_handler.py

```diff
@@ -178,11 +216,18 @@ def handle_single_input(path, output_dir, is_root_outdir, naming_style, known_co
                     if "AMP-binding" in prediction:
                         # ctg_id, orf_idx, a_idx = __parse_amp_binding_domain(prediction)
                         orf_id, a_idx = __parse_amp_binding_domain(prediction)
-                        stachelhaus_seq = domain_predictions[prediction]["NRPSPredictor2"]["stachelhaus_seq"].upper()
+                        new_version = 'nrpys' in domain_predictions[prediction]  # for antismash7. TODO: rewrite all this to treat different versions more carefully
```

Comment:
`new_version` is quite a confusing name.
Also, I suggest doing all the "dirty" work of handling different antiSMASH versions in one place -- SVM_entry class initiation. In this case, all code here will be simply
```
  svm = SVM_entry(domain_predictions[prediction])   # let him decide whether it is a "new version" or not
  parsed_predictions.append({.... , "signature": svm.stachelhaus_seq, "svm": svm})
```